Update the README (#365)
napari/npe2: README.md

# npe2 - napari plugin engine version 2

[![CI](https://github.com/napari/npe2/actions/workflows/ci.yml/badge.svg)](https://github.com/napari/npe2/actions/workflows/ci.yml)
[![codecov](https://codecov.io/gh/napari/npe2/branch/main/graph/badge.svg?token=FTH635x542)](https://codecov.io/gh/napari/npe2)

## Project description

The **napari plugin engine version 2**, **npe2** extends the functionality of
[napari's core](https://github.com/napari/napari).
The plugin ecosystem offers user additional functionality for napari as well
as specific support for different scientific domains.

> This repo contains source code and documentation about the napari plugin engine.

Reviewer: ```suggestion
This repo contains and contains all source code and documentation required for defining, validating and managing plugins for napari.
```

Author: Love it!

## Getting started

The [napari plugin docs landing page](https://napari.org/stable/plugins/index.html)
offers comprehensive information for **plugin users** and for **plugin developers**.

### Plugin users

For plugin users, the docs include information about:
- [Starting to use plugins](https://napari.org/stable/plugins/start_using_plugins/index.html#plugins-getting-started)
- [Finding and installing plugins](https://napari.org/stable/plugins/start_using_plugins/finding_and_installing_plugins.html#find-and-install-plugins)

### Plugin developers

For plugin developers, the docs cover topics like:
- [Building a plugin](https://napari.org/stable/plugins/building_a_plugin/index.html)

> - [Contributing a plugin](https://napari.org/stable/plugins/building_a_plugin/guides.html)

Reviewer: ```suggestion
- [Guides to different plugin contributions](https://napari.org/stable/plugins/building_a_plugin/guides.html)
```

- [Technical references such as the plugin manifest](https://napari.org/stable/plugins/technical_references/manifest.html)

Try the [**napari plugin template**](https://github.com/napari/napari-plugin-template)
to streamline development of a new plugin.

## Installation

> The `npe2` command line tool can be installed with `pip` or `conda`.

Reviewer 2: ```suggestion
The `npe2` command line tool can be installed with `pip` or `conda`, but will already be installed as a dependency if you have napari installed.
```
Perhaps worth adding this?

### Using pip

> 1. Create and activate a virtual environment.

Author: ```suggestion
1. Create and activate a virtual environment. If you are new to using virtual environments, 
    visit our [virtual environments guide](https://napari.org/stable/plugins/virtual_environment_docs/1-virtual-environments.html)
```

>     ```bash

Reviewer: Do we think it's better to link here to our virtual environments guide: https://napari.org/stable/plugins/virtual_environment_docs/1-virtual-environments.html

I can see how we might just want a quickstart so feel free to take or leave this suggestion!

Author: Let's do this. Let's add the documentation link below the quickstart commands.

In general, I like to keep all installation steps together instead of linking out to other docs. Installation is the first experience users have with the repo, and we want them to be successful.

Author: Added an italicized note to with the suggested guide. 👍🏼

- [Building a plugin](https://napari.org/stable/plugins/building_a_plugin/index.html)
- [Guides to different plugin contributions](https://napari.org/stable/plugins/building_a_plugin/guides.html)
- [Technical references such as the plugin manifest](https://napari.org/stable/plugins/technical_references/manifest.html)

Try the [**napari plugin template**](https://github.com/napari/napari-plugin-template)
to streamline development of a new plugin.

## Installation

The `npe2` command line tool can be installed with `pip` or `conda`, but will already be installed as a dependency if you have napari installed.

### Using pip

1. Create and activate a virtual environment.

*If you are new to using virtual environments, visit our [virtual environments guide](https://napari.org/stable/plugins/virtual_environment_docs/1-virtual-environments.html)*.

    ```bash
    python3 -m venv .venv
    source .venv/bin/activate
    ```

2. Install npe2.

    ```bash
    pip install npe2
    ```

3. Test your installation.

    ```bash
    npe2 --help
    ```

### Using conda

1. Create and activate a virtual environment.

    ```bash
    conda create -n npe-test -c conda-forge python=3.12

>     condaa activate npe-test

Reviewer 2: ```suggestion
    conda activate npe-test
```
